### PR TITLE
[BREAKING] Fix skydb skylinks

### PR DESCRIPTION
**Pull request by m-cat**
- [BREAKING] Add sia: prefix to skylinks returned from SkyDB and MySky
- [BREAKING] Add // to URI prefixes (e.g. sia: -> sia://) 
- [BREAKING] Remove uriHandshakeResolverPrefix

### CODE REVIEW

**Review by dghelm — DISMISSED** [review]
Nice cleanup to match conventions. I've never seen `hns://` used before, I'll keep an eye out for its usage.

File: src/utils/url.ts [hunk 1]
```diff
@@ -6,9 +6,9 @@ import { throwValidationError, validateString } from "./validation";
 
 export const defaultSkynetPortalUrl = "https://siasky.net";
 
-export const uriHandshakePrefix = "hns:";
-export const uriHandshakeResolverPrefix = "hnsres:";
-export const uriSkynetPrefix = "sia:";
+export const uriHandshakePrefix = "hns://";
+export const uriHandshakeResolverPrefix = "hnsres://";
```
Comment (kwypchlo) [inline src/utils/url.ts:10]: that shouldn't be a thing - is this used anywhere?

**Review by dghelm — DISMISSED** [review]

**Review by dghelm — APPROVED** [review]

File: src/crypto.ts [hunk 2]
```diff
@@ -125,12 +125,7 @@ export function hashRegistryEntry(registryEntry: RegistryEntry, hashedDataKeyHex
     dataKeyBytes = hashDataKey(registryEntry.dataKey);
   }
 
-  let dataBytes;
-  if (typeof registryEntry.data === "string") {
-    dataBytes = encodeUtf8String(registryEntry.data);
-  } else {
-    dataBytes = encodePrefixedBytes(registryEntry.data);
-  }
+  const dataBytes = encodePrefixedBytes(registryEntry.data);
```
Comment (ro-tex) [inline src/crypto.ts:128]: Q: Just so I understand, are we treating binary and string data the same now or do we expect to only get binary data?

File: src/utils/string.ts [hunk 3]
```diff
@@ -76,14 +76,14 @@ export function trimSuffix(str: string, suffix: string, limit?: number): string
  * @returns - The processed string.
  */
 export function trimUriPrefix(str: string, prefix: string): string {
-  const longPrefix = `${prefix}//`;
-  if (str.startsWith(longPrefix)) {
+  const shortPrefix = trimSuffix(prefix, "/");
+  if (str.startsWith(prefix)) {
     // longPrefix is exactly at the beginning
```
Comment (ro-tex) [inline src/utils/string.ts:81]: super nit: 
```suggestion:
    // prefix is exactly at the beginning
```

**Review by ro-tex — APPROVED** [review]

File: src/crypto.ts (hunk 2 above)
Comment (dghelm) [reply src/crypto.ts:128]: For the registry, we're using binary. This was triggered by confusion between Skylinks as Base64 representation of binary vs string stored as binary on the registry.

Comment (m-cat) [reply src/crypto.ts:128]: The entry data can only be bytes now, yeah.

**Pull request merged**